### PR TITLE
Remove shadowed declarations, undefined identifiers, and specify argument types where it is required.

### PR DESCRIPTION
**Pull request by robertsipka**

### CODE REVIEW

File: jerry-main/main-unix.c [hunk 1]
```diff
@@ -627,7 +627,7 @@ main (int argc,
 
   if (jerry_is_feature_enabled (JERRY_FEATURE_SNAPSHOT_EXEC))
   {
-    for (int i = 0; i < exec_snapshots_count; i++)
+    for (i = 0; i < exec_snapshots_count; i++)
```
Comment (LaszloLango) [inline jerry-main/main-unix.c]: I thinks won't be right. You will override the `i` of the outer `for` statement here. This is unsafe and might lead to infinite loop. Please rename the variable instead, eg.: `for (int j = 0; j < exec_snapshots_count; j++)`

File: jerry-main/main-unix.c [hunk 2]
```diff
@@ -652,7 +652,7 @@ main (int argc,
 
   if (!jerry_value_has_error_flag (ret_value))
   {
-    for (int i = 0; i < files_counter; i++)
```
Comment (LaszloLango) [inline jerry-main/main-unix.c]: ditto

File: jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c [hunk 3]
```diff
@@ -783,10 +783,10 @@ ecma_builtin_array_prototype_object_shift (ecma_value_t this_arg) /**< this argu
     if (ecma_is_value_empty (ret_value))
     {
       len--;
-      ecma_string_t *index_str_p = ecma_new_ecma_string_from_uint32 (len);
+      ecma_string_t *idx_str_p = ecma_new_ecma_string_from_uint32 (len);
```
Comment (zherczeg) [inline jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c]: This seems to be a len_str_p

File: jerry-core/ecma/builtin-objects/ecma-builtin-date.c [hunk 4]
```diff
@@ -319,7 +319,6 @@ ecma_builtin_date_parse (ecma_value_t this_arg, /**< this argument */
       else if (date_str_curr_p < date_str_end_p
                && (*date_str_curr_p == '+' || *date_str_curr_p == '-'))
       {
-        ecma_length_t remaining_length;
         remaining_length = lit_utf8_string_length (date_str_curr_p,
```
Comment (zherczeg) [inline jerry-core/ecma/builtin-objects/ecma-builtin-date.c]: This should have a different name.

File: jerry-main/main-unix.c (hunk 1 above)
Comment (robertsipka) [inline jerry-main/main-unix.c]: That for loop ends before this statement, I've just reuse the `i` variable here.

Comment (galpeter) [inline jerry-main/main-unix.c]: Can't we just move the top level `i` into the required `for`'s init list?

**Comment by robertsipka** [comment]
Thanks, I've updated this patch based on your suggestions.

File: jerry-core/ecma/builtin-objects/ecma-builtin-json.c [hunk 5]
```diff
@@ -934,11 +934,11 @@ ecma_builtin_json_stringify (ecma_value_t this_arg, /**< 'this' argument */
                                    array_length,
                                    ret_value);
 
-      uint32_t array_length = ecma_number_to_uint32 (array_length_num);
+      uint32_t conv_array_length_num = ecma_number_to_uint32 (array_length_num);
       uint32_t index = 0;
 
       /* 4.b.ii */
-      while ((index < array_length) && ecma_is_value_empty (ret_value))
+      while ((index < conv_array_length_num) && ecma_is_value_empty (ret_value))
```
Comment (zherczeg) [inline jerry-core/ecma/builtin-objects/ecma-builtin-json.c]: If this is used only once, we could put the ecma_number_to_uint32 (array_length_num) here.

File: jerry-main/main-unix.c [hunk 6]
```diff
@@ -744,7 +743,7 @@ main (int argc,
     const char *prompt = !no_prompt ? "jerry> " : "";
     bool is_done = false;
 
-    jerry_value_t global_obj_val = jerry_get_global_object ();
+    global_obj_val = jerry_get_global_object ();
```
Comment (zherczeg) [inline jerry-main/main-unix.c]: I think this value is initialized to jerry_get_global_object (). Is there no reference counting problem here?

File: jerry-core/vm/vm.c [hunk 7]
```diff
@@ -753,9 +753,9 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
           {
             case VM_OC_GET_LITERAL_LITERAL:
             {
-              uint16_t literal_index;
-              READ_LITERAL_INDEX (literal_index);
-              READ_LITERAL (literal_index, right_value);
+              uint16_t literal_literal_index;
```
Comment (zherczeg) [inline jerry-core/vm/vm.c]: second_literal_index

File: jerry-core/parser/js/js-parser-util.c [hunk 8]
```diff
@@ -192,9 +192,9 @@ parser_flush_cbc (parser_context_t *context_p) /**< context */
       {
         literal_index = context_p->last_cbc.third_literal_index;
 
-        lexer_literal_t *literal_p = PARSER_GET_LITERAL (literal_index);
+        lexer_literal_t *third_literal_p = PARSER_GET_LITERAL (literal_index);
```
Comment (zherczeg) [inline jerry-core/parser/js/js-parser-util.c]: We should reuse literal_p since literal_index is also reused.

File: jerry-main/main-unix.c (hunk 6 above)
Comment (robertsipka) [inline jerry-main/main-unix.c]: It's freed earlier, but I've renamed it to avoid the misleading.

**Pull request merged**